scanpipe/apps.py

```diff
@@ -20,45 +20,51 @@
 # ScanCode.io is a free software code scanning tool from nexB Inc. and others.
 # Visit https://github.com/nexB/scancode.io for support and download.
 
-from pathlib import Path
-
 from django.apps import AppConfig
+from django.core.exceptions import ImproperlyConfigured
 from django.utils.translation import gettext_lazy as _
 
-dot_py_suffix = ".py"
-
+try:
+    from importlib import metadata as importlib_metadata
+except ImportError:
+    import importlib_metadata
 
-def remove_dot_py_suffix(filename):
-    """
-    Return the `filename` without the trailing ".py" suffix if any.
-    """
-    if filename.endswith(dot_py_suffix):
-        return filename[: -len(dot_py_suffix)]
-    return filename
+from scanpipe.pipelines import is_pipeline_subclass
 
 
 class ScanPipeConfig(AppConfig):
     name = "scanpipe"
     verbose_name = _("ScanPipe")
-    pipelines = []
+    pipelines = {}
 
     def ready(self):
+        self.load_pipelines()
+
+    def load_pipelines(self):
         """
-        Load available Pipelines.
+        Load Pipelines from the "scancodeio_pipelines" entry point group.
         """
-        project_root = Path(__file__).parent.parent.absolute()
-        pipelines_dir = project_root / "scanpipe" / "pipelines"
+        entry_points = importlib_metadata.entry_points()
 
-        for child in pipelines_dir.iterdir():
-            if child.name.endswith(dot_py_suffix) and not child.name.startswith("_"):
-                location = str(child.relative_to(project_root))
-                name = remove_dot_py_suffix(child.name)
-                self.pipelines.append((location, name))
+        # Ignore duplicated entries caused by duplicated paths in `sys.path`.
```

Comment:
1. are duplicated entries possible?
2. should we instead raise an exception and fail to avoid any possible confusion when this happens? ... especially since below you are also raising exceptions https://github.com/nexB/scancode.io/pull/95/files#diff-450348e775480cb66b78166191948b47e5e49d1a97563f22a219269b4f977071R65